modules/http4s-emitter/src/main/scala/com.snowplowanalytics.snowplow.scalatracker.emitters.http4s/Http4sEmitter.scala

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2020-2020 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.scalatracker.emitters.http4s
+
+import cats.{Monad, MonadError}
+import cats.implicits._
+import cats.effect.{Concurrent, Fiber, Resource, Sync, Timer}
+import fs2.{Pipe, Stream}
+import fs2.concurrent.{Dequeue, Enqueue, Queue}
+import org.http4s.client.Client
+import org.http4s.headers.`Content-Type`
+import org.http4s.{MediaType, Method, Request => HttpRequest, Uri}
+import scala.concurrent.duration._
+import scala.util.Random
+
+import com.snowplowanalytics.snowplow.scalatracker.{Buffer, Emitter, Payload}
+import com.snowplowanalytics.snowplow.scalatracker.Emitter._
+import com.snowplowanalytics.snowplow.scalatracker.Buffer.Action
+
+object Http4sEmitter {
+
+  def build[F[_]: Concurrent: Timer](collector: EndpointParams,
+                                     client: Client[F],
+                                     bufferConfig: BufferConfig    = BufferConfig.Default,
+                                     retryPolicy: RetryPolicy      = RetryPolicy.Default,
+                                     queuePolicy: EventQueuePolicy = EventQueuePolicy.Default,
+                                     callback: Option[Callback[F]] = None): Resource[F, Emitter.Flushable[F]] =
+    Resource {
+      for {
+        queue <- queueForPolicy[F](queuePolicy)
+        fiber <- Concurrent[F].start(drainQueue(queue, client, collector, bufferConfig, retryPolicy, callback))
+      } yield instance(queue) -> shutdown(fiber, queue)
+    }
+
+  private def instance[F[_]](queue: Enqueue[F, Action]): Emitter.Flushable[F] =
+    new Emitter.Flushable[F] {
+      override def send(event: Payload): F[Unit] =
+        queue.enqueue1(Action.Enqueue(event))
+
+      override def flushBuffer(): F[Unit] =
+        queue.enqueue1(Action.Flush)
+    }
+
+  private def queueForPolicy[F[_]: Concurrent](policy: EventQueuePolicy): F[Queue[F, Action]] =
+    policy match {
+      case EventQueuePolicy.UnboundedQueue      => Queue.unbounded
+      case EventQueuePolicy.BlockWhenFull(max)  => Queue.bounded(max)
+      case EventQueuePolicy.IgnoreWhenFull(max) => Queue.bounded(max)
```

Comment:
If we're happy with `bounded` everywhere it can be:

```scala
  private def queueForPolicy[F[_]: Concurrent](policy: EventQueuePolicy): F[Queue[F, Action]] =
    policy.tryLimit match {
      case None      => Queue.unbounded
      case Some(max) => Queue.bounded(max)
    }
```

Reply:
Also it seems we don't actually use it in `Http4sEmitter`?